firestore/google/cloud/firestore_v1/query.py

```diff
@@ -968,3 +985,32 @@ def _query_response_to_snapshot(response_pb, collection, expected_prefix):
         update_time=response_pb.document.update_time,
     )
     return snapshot
+
+
+def _collection_group_query_response_to_snapshot(response_pb, collection):
+    """Parse a query response protobuf to a document snapshot.
+
+    Args:
+        response_pb (google.cloud.proto.firestore.v1.\
+            firestore_pb2.RunQueryResponse): A
+        collection (~.firestore_v1.collection.CollectionReference): A
+            reference to the collection that initiated the query.
+
+    Returns:
+        Optional[~.firestore.document.DocumentSnapshot]: A
+        snapshot of the data returned in the query. If ``response_pb.document``
+        is not set, the snapshot will be :data:`None`.
+    """
+    if not response_pb.HasField("document"):
+        return None
```

Comment:
Does Python not return empty DocumentSnapshots for non-existing documents? In either case, this is a query and this condition should never trigger (fingers crossed).

Reply:
@schmidt-sebastian [`DocumentReference.get`](https://github.com/googleapis/google-cloud-python/blob/3618354c5fdd388e06b0b419cbcbe613cb6ad879/firestore/google/cloud/firestore_v1/document.py#L408-L468) does return an empty snapshot for a non-existent document.  For the query case, the back-end can return a response without a `document` when reporting partial results:  see #6924 and PR #7206 for the history.

Reply:
My bad: It looks like the expected behavior is that it will get skipped (see https://cs.corp.google.com/piper///depot/google3/google/firestore/v1beta1/firestore.proto?q=runqueryrequest+firestore&dr=CSs&l=619)